src/repository/storage/postgres/orm/sequelize/noteRelations.ts

```diff
@@ -209,4 +210,36 @@ export default class NoteRelationsSequelizeStorage {
 
     return foundNote !== null;
   };
+
+  /**
+   * Get all parent notes of a note that a user has access to,
+   * where the user has access to.
+   * @param noteId - the ID of the note.
+   */
+  public async getNoteParentsIds(noteId: NoteInternalId): Promise<NoteInternalId[]> {
+    // get all note ids via a singe sql query instead of many
+    const query = `
```

Comment:
Please, add some comments to logic of this query

Reply:
@slaveeks can you check now?